% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
%
\documentclass[
]{report}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{titlesec, blindtext, color, float}

\titleformat{\chapter}[display]
  {\Huge\bfseries}
  {}
  {0pt}
  {\thechapter.\ }

\titleformat{name=\chapter,numberless}[display]
  {\Huge\bfseries}
  {}
  {0pt}
  {}

\titlespacing*{\chapter}{0pt}{0pt}{40pt}

\renewcommand{\bibname}{References}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage[]{natbib}
\bibliographystyle{apalike}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={Ovarian Cancer Histotypes: Report of Statistical Findings},
  pdfauthor={Derek Chiu},
  colorlinks=true,
  linkcolor={Maroon},
  filecolor={Maroon},
  citecolor={Blue},
  urlcolor={Blue},
  pdfcreator={LaTeX via pandoc}}

\title{Ovarian Cancer Histotypes: Report of Statistical Findings}
\author{Derek Chiu}
\date{2024-07-16}

\begin{document}
\maketitle

{
\hypersetup{linkcolor=}
\setcounter{tocdepth}{1}
\tableofcontents
}
\listoffigures
\listoftables
\chapter*{Preface}\label{preface}
\addcontentsline{toc}{chapter}{Preface}

This report of statistical findings describes the classification of ovarian cancer histotypes using data from NanoString CodeSets.

Marina Pavanello conducted the initial exploratory data analysis, Cathy Tang implemented class imbalance techniques, Derek Chiu conducted the normalization and statistical analysis, and Lauren Tindale and Aline Talhouk are the project leads.

\chapter{Introduction}\label{introduction}

Ovarian cancer has five major histotypes: high-grade serous carcinoma (HGSC), low-grade serous carcinoma (LGSC), endometrioid carcinoma (ENOC), mucinous carcinoma (MUC), and clear cell carcinoma (CCOC). A common problem with classifying these histotypes is that there is a class imbalance issue. HGSC dominates the distribution, commonly accounting for 70\% of cases in many patient cohorts, while the other four histotypes are spread over the rest of the cases. Subsampling methods like up-sampling, down-sampling, and SMOTE can be used to mitigate this problem.

The supervised learning is performed under a consensus framework: we consider various classification algorithms and use evaluation metrics like accuracy, F1-score, Kappa, and G-mean to inform the decision of which methods to carry forward for prediction in confirmation and validation sets.

\chapter{Methods}\label{methods}

\section{Pre-Processing}\label{pre-processing}

\subsection{Case Selection}\label{case-selection}

Raw data comes from three NanoString CodeSets (CS): CS1, CS2, and CS3. We divide the data into training, confirmation, and validation sets by using samples from these sets of cohorts:

\begin{itemize}
\tightlist
\item
  Training

  \begin{itemize}
  \tightlist
  \item
    CS1: MAYO, OOU, OOUE, VOA, MTL
  \item
    CS2: MAYO, OOU, OOUE, OVAR3, VOA, ICON7, JAPAN, MTL, POOL-CTRL
  \item
    CS3: OOU, OOUE, VOA, POOL-1, POOL-2, POOL-3
  \end{itemize}
\item
  Confirmation:

  \begin{itemize}
  \tightlist
  \item
    CS3: TNCO
  \end{itemize}
\item
  Validation:

  \begin{itemize}
  \tightlist
  \item
    CS3: DOVE4
  \end{itemize}
\end{itemize}

\subsection{Quality Control}\label{quality-control}

Samples that failed any of the following NanoString quality control conditions were removed:

\begin{itemize}
\tightlist
\item
  \texttt{linFlag}: linearity of positive controls with positive control concentrations is less than 0.95, or linearity measures are unknown
\item
  \texttt{imagingFlag}: percent of field of view is less than 75\%
\item
  \texttt{spcFlag}: smallest positive control is less than the lower limit of detection (negative control average expression less two times the negative control standard deviation), or negative control average expression equals zero
\item
  \texttt{normFlag}: signal to noise ratio less than 100, or percent of genes detected is less than 50. Note: these thresholds were determined by examining the \hyperref[gd-vs.-snr]{\%GD vs.~SNR} relationship below.
\end{itemize}

\subsection{Normalization}\label{normalization}

The full training set (n=1243) is comprised of data from CodeSets (CS) 1, 2, and 3. All CodeSets were first normalized to housekeeping genes, then different approaches were taken for subsequent normalizations of each CodeSet.

CS1 was normalized to CS3 using five ``Random1'' reference samples. These reference samples are randomly selected from CS1 among all samples in the three CodeSets that share common otta IDs, such that we obtain one sample from each of the five histotypes. Then, we use the reference-based method to normalize CS1 to CS3 across their common genes, for the remaining expression samples \citet{talhouk2016}.

Similarly, CS2 was normalized to CS3 using the same ``Random1'' reference samples, now taken from CS2. Normalization was performed across common genes between CS2 and CS3.

For CS3, we first split the dataset into three sites: Vancouver, USC, and AOC. We use the CS3-Vancouver subset as a ``reference standard'', and normalized CS3-USC and CS3-AOC to CS3-Vancouver using a ``Random1'' reference set randomly selected among samples common between Vancouver, USC, and AOC. Finally, the CS3-Vancouver expression samples are included in the training set without further normalization.

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/venn-genes-1} 

}

\caption{Venn diagram of common and unique gene targets covered by each CodeSet}\label{fig:venn-genes}
\end{figure}

\subsection{Final Processing}\label{final-processing}

We map ovarian histotypes to all remaining samples and keep the major histotypes for building the predictive model: high-grade serous carcinoma (HGSC), clear cell ovarian carcinoma (CCOC), endometrioid ovarian carcinoma (ENOC), low-grade serous carcinoma (LGSC), mucinous carcinoma (MUC).

Duplicate cases (two samples with the same ottaID) were removed before generating the final training set to use for fitting the classification models. All CS3 cases were preferred over CS1 and CS2, and CS3-Vancouver cases were preferred over CS3-AOC and CS3-USC when selecting duplicates.

The final training set used only genes that were common across all three CodeSets.

\section{Classifiers}\label{classifiers}

We use 4 classification algorithms in the supervised learning framework for the Training Set. The pipeline was run using SLURM batch jobs submitted to a partition on a CentOS 7 server. All resampling techniques, pre-processing, model specification, hyperparameter tuning, and evaluation metrics were implemented using the \texttt{tidymodels} suite of packages. The classifiers we used are:

\begin{itemize}
\tightlist
\item
  Random Forest (\texttt{rf})
\item
  Support Vector Machine (\texttt{svm})
\item
  XGBoost (\texttt{xgb})
\item
  Regularized Multinomial Regression (\texttt{mr})
\end{itemize}

\subsection{Resampling of Training Set}\label{resampling-of-training-set}

We used a nested cross-validation design to assess each classifier while also performing hyperparameter tuning. An outer 5-fold CV stratified by histotype was used together with an inner 5-fold CV with 2 repeats stratified by histotype. This design was chosen such that the test sets of the inner resamples would still have a reasonable number of samples belonging to the smallest minority class.

The outer resampling method cannot be the bootstrap, because the inner training and inner test sets will likely contain the same samples as a result of sampling with replacement in the outer training set. This phenomenon might result in inflated performance as some observations are used both to train and evaluate the hyperparameter tuning in the inner loop.

\subsection{Hyperparameter Tuning}\label{hyperparameter-tuning}

The following specifications for each classifier were used for tuning hyperparameters:

\begin{itemize}
\tightlist
\item
  \texttt{rf} and \texttt{xgb}: The number of trees were fixed at 500. Other hyperparameters were tuned across 10 randomly selected points in a latin hypercube design.
\item
  \texttt{svm}: Both the cost and sigma hyperparameters were tuned across 10 randomly selected points in a latin hypercube design. We tuned the cost parameter in the range {[}1, 8{]}. The range for tuning the sigma parameter was obtained from the 10\% and 90\% quantiles of the estimation using the \texttt{kernlab::sigest()} function.
\item
  \texttt{mr}: We generated 10 randomly selected points in a latin hypercube design for the penalty (lambda) parameter. Then, we generated 10 evenly spaced points in {[}0, 1{]} for the mixture (alpha) parameter in the regularized multinomial regression model. These two sets of 10 points were crossed to generate a tuning grid of 100 points.
\end{itemize}

The hyperparameter combination that resulted in the highest average F1-score across the inner training sets was selected for each classifier to use as the model for assessing prediction performance in the outer training loop.

\subsection{Subsampling}\label{subsampling}

Here are the specifications of the subsampling methods used to handle class imbalance:

\begin{itemize}
\tightlist
\item
  None: No subsampling is performed
\item
  Down-sampling: All levels except the minority class are sampled down to the same frequency as the minority class
\item
  Up-sampling: All levels except the majority class are sampled up to the same frequency as the majority class
\item
  SMOTE: All levels except the majority class have synthetic data generated until they have the same frequency as the majority class
\item
  Hybrid: All levels except the majority class have synthetic data generated up to 50\% of the frequency of the majority class, then the majority class is sampled down to the same frequency as the rest.
\end{itemize}

The figure below helps visualize how the distribution of classes changes when we apply subsampling techniques to handle class imbalance:

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/sampling-1} 

}

\caption{Visualization of Subsampling Techniques}\label{fig:sampling}
\end{figure}

\subsection{Workflows}\label{workflows}

The 4 \textbf{algorithms} and 5 \textbf{subsampling} methods are crossed to create 20 different classification \textbf{workflows}. For example, the \texttt{hybrid\_xgb} workflow is a classifier that first pre-processes a training set by applying a hybrid subsampling method, and then proceeds to use the XGBoost algorithm to classify ovarian histotypes.

\section{Two-Step Algorithm}\label{two-step-algorithm}

The HGSC histotype comprises of approximately 80\% of cases among ovarian carcinoma patients, while the remaining 20\% of cases are relatively, evenly distributed among ENOC, CCOC, LGSC, and MUC histotypes. We can implement a two-step algorithm as such:

\begin{itemize}
\tightlist
\item
  Step 1: use binary classification for HGSC vs.~non-HGSC
\item
  Step 2: use multinomial classification for the remaining non-HGSC classes
\end{itemize}

Let

\[
\begin{aligned}
& X_k = \text{Training data with k classes}  \\
& C_k = \text{Class with highest}\;F_1\;\text{score from training}\;X_k \\
& W_k = \text{Workflow associated with}\;C_k
\end{aligned}
\label{eq:sequential}
\]

Figure \ref{fig:two-step-flowchart} shows how the two-step algorithm works:

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/two-step-flowchart-1} 

}

\caption{Two-Step Algorithm}\label{fig:two-step-flowchart}
\end{figure}

Although the class imbalance problem is mostly eliminated in Step 2 after removing the HGSC cases, we still use the same subsampling method in Step 2 as was used in Step 1 to keep the algorithm consistent.

\subsection{Aggregating Predictions}\label{aggregating-predictions}

The aggregation for two-step predictions is quite straightforward:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Predict HGSC vs.~non-HGSC
\item
  Among all non-HGSC cases, predict CCOC vs.~LGSC vs.~MUC vs.~ENOC
\end{enumerate}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/two-step-predictions-1} 

}

\caption{Aggregating Predictions for Two-Step Algorithm}\label{fig:two-step-predictions}
\end{figure}

\section{Sequential Algorithm}\label{sequential-algorithm}

Instead of training on k classes simultaneously using multinomial classifiers, we can use a sequential algorithm that performs k-1 one-vs-all binary classifications iteratively to obtain a final prediction of all cases. At each step in the sequence, we classify one class vs.~all other classes, where the classes that make up the ``other'' class are those not equal to the current ``one'' class and excluding all ``one'' classes from previous steps. For example, if the ``one'' class in step 1 was HGSC, the ``other'' classes would include CCOC, ENOC, LGSC, and MUC. If the ``one'' class in step 2 was CCOC, the ``other'' classes include ENOC, LGSC, and MUC.

The order of classes and workflows to use at each step in the sequential algorithm must be determined using a retraining procedure. After removing the data associated with a particular class, we retrain using the remaining data using multinomial classifiers as described before. The class and workflow to use for the next step in the sequence is selected based on the best per-class evaluation metric value (e.g.~F1-score).

Figure \ref{fig:sequential-flowchart} illustrates how the sequential algorithm works for K=5, using ovarian histotypes as an example for the classes.

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/sequential-flowchart-1} 

}

\caption{Sequential Algorithm}\label{fig:sequential-flowchart}
\end{figure}

The subsampling method used in the first step of the sequential algorithm is used in all subsequent steps in order to maintain data pre-processing consistency. As a result, we are only comparing classification algorithms within one subsampling method across the entire sequential algorithm.

\subsection{Aggregating Predictions}\label{aggregating-predictions-1}

We have to aggregate the one-vs-all predictions from each of the sequential algorithm workflows in order to obtain a final class prediction on a holdout test set. Each sequential workflow has to be assessed on every sample to ensure that cases classified into the ``all'' class from a previous step of the sequence are eventually assigned a predicted class. For example, say that based on certain class-specific metrics we determined that the order of classes in the sequential algorithm was to predict HGSC vs.~non-HGSC, CCOC vs.~non-CCOC, LGSC vs.~non-LGSC, and then MUC vs.~ENOC. Figure \ref{fig:sequential-predictions} illustrates how the final predictions are assigned:

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/sequential-predictions-1} 

}

\caption{Aggregating Predictions for Sequential Algorithm}\label{fig:sequential-predictions}
\end{figure}

\section{Gene Optimization}\label{gene-optimization}

We want to discover an optimal set of genes for the classifiers while including specific genes from other studies such as PrOTYPE and SPOT. A total of 72 genes are used in the classifier training set.

There are 16 genes in the classifier set that overlap with the PrOTYPE classifier: COL11A1, CD74, CD2, TIMP3, LUM, CYTIP, COL3A1, THBS2, TCF7L1, HMGA2, FN1, POSTN, COL1A2, COL5A2, PDZK1IP1, FBN1.

There are also 13 genes in the classifier set that overlap with the SPOT signature: HIF1A, CXCL10, DUSP4, SOX17, MITF, CDKN3, BRCA2, CEACAM5, ANXA4, SERPINE1, TCF7L1, CRABP2, DNAJC9.

We obtain a total of 28 genes from the union of PrOTYPE and SPOT genes that we want to include in the final classifier, regardless of model performance. We then incrementally add genes one at a time from the remaining 44 candidate genes based on an overall variable importance rank to the set of 28 base genes and recalculate performance metrics. The number of genes at which the performance peaks or starts to plateau may indicate an optimal gene set model for us to compare with the full set model.

\subsection{Variable Importance}\label{variable-importance}

Variable importance is calculated using either a model-based approach if it is available, or a permutation-based VI score otherwise. The variable importance scores are averaged across the outer training folds, and then ranked from highest to lowest.

For the sequential and two-step classifiers, we calculate an overall VI rank by taking the cumulative union of genes at each variable importance rank across all sequences, until all genes have been included.

The variable importance measures are:

\begin{itemize}
\item
  Random Forest: impurity measure (Gini index)
\item
  XGBoost: gain (fractional contribution of each feature to the model based on the total gain of the corresponding features's splits)
\item
  SVM: permutation based p-values
\item
  Multinomial regression: absolute value of estimated coefficients at cross-validated lambda value
\end{itemize}

\section{Performance Evaluation}\label{performance-evaluation}

\subsection{Class Metrics}\label{class-metrics}

We use the accuracy, sensitivity, specificity, F1-score, kappa, balanced accuracy, and geometric mean, as class metrics to measure both training and test performance between different workflows. Multiclass extensions of these metrics can be calculated except for F1-score, where we use macro-averaging to obtain an overall metric. Class-specific metrics are calculated by recoding classes into one-vs-all categories for each class.

\subsubsection{Accuracy}\label{accuracy}

The accuracy is defined as the proportion of correct predictions out of all cases:

\[
\text{accuracy} = \frac{TP}{TP + FP + FN + TN}
\label{eq:accuracy}
\]

\subsubsection{Sensitivity}\label{sensitivity}

Sensitivity is the proportional of correctly predicted positive cases, out of all cases that were truly positive

\[
\text{sensitivity} = \frac{TP}{TP + FN}
\label{eq:sens}
\]

\subsubsection{Specificity}\label{specificity}

Specificity is the proportional of correctly predicted negative cases, out of all cases that were truly negative.

\[
\text{specificity} = \frac{TN}{TN + FP}
\label{eq:spec}
\]

\subsubsection{F1-Score}\label{f1-score}

The F-measure can be thought of as a harmonic mean between precision and recall:

\[
F_{meas} = \frac{(1 + \beta^2) \times precision \times recall}{(\beta^2 \times precision) + recall}
\label{eq:f1}
\]

The \(\beta\) value can be adjusted to place more weight upon precision or recall. The most common value is \(\beta\) is 1, which is also commonly known as the F1-score. A multiclass extension doesn't exist for the F1-score, so we use macro-averaging to calculate this metric when there are more than two classes. For example, with \(k\) classes, the macro-averaged F1-score is equal to:

\[
{F_1}_{macro} = \frac{1}{k} \sum_{i=1}^{k}{F_1}_{i}
\label{eq:f1-macro}
\] where each \({F_1}_{i}\) is the F1-score computed frrom recoding classes into \(k=i\) vs.~\(k \neq i\).

In situations where there is not at least one predicted case for each of the classes (e.g.~for a poor classifier), \({F_1}_{i}\) is undefined because the per-class precision of class \(i\) is undefined. Those \({F_1}_{i}\) terms are removed from the \({F_1}_{macro}\) equation and the resulting value may be inflated. Interpreting the F1-score in such a case would be misleading.

\subsubsection{Kappa}\label{kappa}

Kappa is the defined as:

\[
\text{kappa} = \frac{p_0 - p_e}{1 - p_e}
\label{eq:kappa}
\]

where \(p_0\) is the observed agreement among raters and \(p_e\) is the hypothetical probability of agreement due to random chance.

\subsubsection{Balanced Accuracy}\label{balanced-accuracy}

Balanced accuracy is the arithmetic mean of sensitivity and specificity.

\[
\text{Balanced Accuracy} = \frac{\text{Sensitivity} + \text{Specificity}}{2}
\label{eq:bal-accuracy}
\]

\subsubsection{Geometric Mean}\label{geometric-mean}

The geometric mean (G-mean) is the \(k^{th}\) root of the product of class-specific sensitivities for \(k\) classes:

\[
\text{G-mean} = \sqrt[k]{\prod_{i=1}^{k}{\text{Sensitivity}_k}}
\label{eq:gmean}
\] The G-mean generalizes easily for the multiclass scenario.

\subsection{AUC}\label{auc}

The area under the receiver operating curve (AUC) is calculated by adding up the area under the curve formed by plotting sensitivity vs.~1 - specificity. The Hand-till method is used as a multiclass extension for the AUC.

We did not use AUC to measure class-specific training set performance because combining predicted probabilities in a one-vs-all fashion might be potentially misleading. The sum of probabilities that add up to the ``other'' class is not equivalent to the predicted probability of the ``other'' class when using a multiclass classifier.

Instead, we only reported ROC curves and their associated AUCs for the test set performance of the sequential and two-step algorithms.

\chapter{Distributions}\label{distributions}

\section{CodeSet Counts}\label{codeset-counts}

\begin{table}

\caption{\label{tab:codeset-counts}Training Set Counts by CodeSet and Processing Stage}
\centering
\begin{tabular}[t]{l|r|r|r|r}
\hline
Processing Stage & CS1 & CS2 & CS3 & Total\\
\hline
Selected Cohorts & 294 & 903 & 2477 & 3674\\
\hline
QC & 286 & 882 & 2273 & 3441\\
\hline
Normalized to Reference & 263 & 827 & 2095 & 3185\\
\hline
Combine CS3 Training with AOC/USC sites & 263 & 827 & 514 & 1604\\
\hline
Major Histotypes & 244 & 779 & 506 & 1529\\
\hline
Deduplicated & 77 & 698 & 468 & 1243\\
\hline
\end{tabular}
\end{table}

\section{Histotype Distribution}\label{histotype-distribution}

\begin{table}

\caption{\label{tab:hist-codeset}Histotype Distribution by CodeSet and Processing Stage}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
Variable & Levels & CS1 & CS2 & CS3 & Total\\
\hline
\multicolumn{6}{l}{\textbf{Selected Cohorts}}\\
\hline
 & HGSC & 126 (43\%) & 655 (73\%) & 1779 (72\%) & 2560 (70\%)\\
\cline{2-6}
 & CCOC & 48 (16\%) & 61 (7\%) & 181 (7\%) & 290 (8\%)\\
\cline{2-6}
 & ENOC & 60 (20\%) & 34 (4\%) & 268 (11\%) & 362 (10\%)\\
\cline{2-6}
 & MUC & 20 (7\%) & 62 (7\%) & 77 (3\%) & 159 (4\%)\\
\cline{2-6}
 & LGSC & 21 (7\%) & 21 (2\%) & 42 (2\%) & 84 (2\%)\\
\cline{2-6}
\multirow{-6}{*}{\raggedright\arraybackslash \hspace{1em}Histotype} & Other & 19 (6\%) & 70 (8\%) & 130 (5\%) & 219 (6\%)\\
\cline{1-6}
\hspace{1em}Total & N (\%) & 294 (8\%) & 903 (25\%) & 2477 (67\%) & 3674 (100\%)\\
\cline{1-6}
\multicolumn{6}{l}{\textbf{QC}}\\
\hline
 & HGSC & 120 (42\%) & 641 (73\%) & 1636 (72\%) & 2397 (70\%)\\
\cline{2-6}
 & CCOC & 48 (17\%) & 61 (7\%) & 173 (8\%) & 282 (8\%)\\
\cline{2-6}
 & ENOC & 60 (21\%) & 32 (4\%) & 229 (10\%) & 321 (9\%)\\
\cline{2-6}
 & MUC & 19 (7\%) & 60 (7\%) & 69 (3\%) & 148 (4\%)\\
\cline{2-6}
 & LGSC & 20 (7\%) & 21 (2\%) & 40 (2\%) & 81 (2\%)\\
\cline{2-6}
\multirow{-6}{*}{\raggedright\arraybackslash \hspace{1em}Histotype} & Other & 19 (7\%) & 67 (8\%) & 126 (6\%) & 212 (6\%)\\
\cline{1-6}
\hspace{1em}Total & N (\%) & 286 (8\%) & 882 (26\%) & 2273 (66\%) & 3441 (100\%)\\
\cline{1-6}
\multicolumn{6}{l}{\textbf{Normalized to Reference}}\\
\hline
 & HGSC & 116 (44\%) & 622 (75\%) & 1540 (74\%) & 2278 (72\%)\\
\cline{2-6}
 & CCOC & 44 (17\%) & 54 (7\%) & 158 (8\%) & 256 (8\%)\\
\cline{2-6}
 & ENOC & 55 (21\%) & 27 (3\%) & 215 (10\%) & 297 (9\%)\\
\cline{2-6}
 & MUC & 15 (6\%) & 57 (7\%) & 58 (3\%) & 130 (4\%)\\
\cline{2-6}
 & LGSC & 14 (5\%) & 19 (2\%) & 35 (2\%) & 68 (2\%)\\
\cline{2-6}
\multirow{-6}{*}{\raggedright\arraybackslash \hspace{1em}Histotype} & Other & 19 (7\%) & 48 (6\%) & 89 (4\%) & 156 (5\%)\\
\cline{1-6}
\hspace{1em}Total & N (\%) & 263 (8\%) & 827 (26\%) & 2095 (66\%) & 3185 (100\%)\\
\cline{1-6}
\multicolumn{6}{l}{\textbf{Combine CS3 Training with AOC/USC sites}}\\
\hline
 & HGSC & 116 (44\%) & 622 (75\%) & 475 (92\%) & 1213 (76\%)\\
\cline{2-6}
 & CCOC & 44 (17\%) & 54 (7\%) & 8 (2\%) & 106 (7\%)\\
\cline{2-6}
 & ENOC & 55 (21\%) & 27 (3\%) & 8 (2\%) & 90 (6\%)\\
\cline{2-6}
 & MUC & 15 (6\%) & 57 (7\%) & 9 (2\%) & 81 \vphantom{1} (5\%)\\
\cline{2-6}
 & LGSC & 14 (5\%) & 19 (2\%) & 6 (1\%) & 39 (2\%)\\
\cline{2-6}
\multirow{-6}{*}{\raggedright\arraybackslash \hspace{1em}Histotype} & Other & 19 (7\%) & 48 (6\%) & 8 (2\%) & 75 (5\%)\\
\cline{1-6}
\hspace{1em}Total & N (\%) & 263 (16\%) & 827 (52\%) & 514 (32\%) & 1604 (100\%)\\
\cline{1-6}
\multicolumn{6}{l}{\textbf{Major Histotypes}}\\
\hline
 & HGSC & 116 (48\%) & 622 (80\%) & 475 (94\%) & 1213 (79\%)\\
\cline{2-6}
 & CCOC & 44 (18\%) & 54 (7\%) & 8 (2\%) & 106 (7\%)\\
\cline{2-6}
 & ENOC & 55 (23\%) & 27 (3\%) & 8 (2\%) & 90 (6\%)\\
\cline{2-6}
 & MUC & 15 (6\%) & 57 (7\%) & 9 (2\%) & 81 (5\%)\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash \hspace{1em}Histotype} & LGSC & 14 (6\%) & 19 (2\%) & 6 (1\%) & 39 (3\%)\\
\cline{1-6}
\hspace{1em}Total & N (\%) & 244 (16\%) & 779 (51\%) & 506 (33\%) & 1529 (100\%)\\
\cline{1-6}
\multicolumn{6}{l}{\textbf{Deduplicated}}\\
\hline
 & HGSC & 9 (12\%) & 552 (79\%) & 451 (96\%) & 1012 (81\%)\\
\cline{2-6}
 & CCOC & 25 (32\%) & 52 (7\%) & 4 (1\%) & 81 (7\%)\\
\cline{2-6}
 & ENOC & 37 (48\%) & 25 (4\%) & 4 (1\%) & 66 (5\%)\\
\cline{2-6}
 & MUC & 3 (4\%) & 53 (8\%) & 5 (1\%) & 61 (5\%)\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash \hspace{1em}Histotype} & LGSC & 3 (4\%) & 16 (2\%) & 4 (1\%) & 23 (2\%)\\
\cline{1-6}
\hspace{1em}Total & N (\%) & 77 (6\%) & 698 (56\%) & 468 (38\%) & 1243 (100\%)\\
\hline
\end{tabular}
\end{table}

\begin{table}

\caption{\label{tab:hist-conf-val}Histotype Distribution in Confirmation and Validation Sets}
\centering
\begin{tabular}[t]{l|l|l|l}
\hline
Variable & Levels & Confirmation & Validation\\
\hline
Histotype & HGSC & 422 (66\%) & 667 (75\%)\\
\hline
 & CCOC & 75 (12\%) & 79 (9\%)\\
\hline
 & ENOC & 106 (16\%) & 105 (12\%)\\
\hline
 & MUC & 27 (4\%) & 26 (3\%)\\
\hline
 & LGSC & 13 (2\%) & 18 (2\%)\\
\hline
Total & N (\%) & 643 (42\%) & 895 (58\%)\\
\hline
\end{tabular}
\end{table}

\section{Cohort Distribution}\label{cohort-distribution}

\begin{table}

\caption{\label{tab:cohorts-dist}Pre-QC Cohort Distribution by CodeSet}
\centering
\begin{tabular}[t]{l|c|c|c}
\hline
**CodeSet** & **CS1**, N = 294 & **CS2**, N = 903 & **CS3**, N = 2,477\\
\hline
Cohort &  &  & \\
\hline
OOU & 108 (37\%) & 43 (4.8\%) & 19 (0.8\%)\\
\hline
OOUE & 32 (11\%) & 30 (3.3\%) & 11 (0.4\%)\\
\hline
VOA & 145 (49\%) & 122 (14\%) & 538 (22\%)\\
\hline
OVAR3 & 0 (0\%) & 150 (17\%) & 0 (0\%)\\
\hline
ICON7 & 0 (0\%) & 416 (46\%) & 0 (0\%)\\
\hline
MAYO & 6 (2.0\%) & 63 (7.0\%) & 0 (0\%)\\
\hline
DOVE4 & 0 (0\%) & 0 (0\%) & 1,160 (47\%)\\
\hline
TNCO & 0 (0\%) & 0 (0\%) & 691 (28\%)\\
\hline
MTL & 3 (1.0\%) & 59 (6.5\%) & 0 (0\%)\\
\hline
JAPAN & 0 (0\%) & 8 (0.9\%) & 0 (0\%)\\
\hline
POOL-CTRL & 0 (0\%) & 12 (1.3\%) & 0 (0\%)\\
\hline
POOL-1 & 0 (0\%) & 0 (0\%) & 31 (1.3\%)\\
\hline
POOL-2 & 0 (0\%) & 0 (0\%) & 14 (0.6\%)\\
\hline
POOL-3 & 0 (0\%) & 0 (0\%) & 13 (0.5\%)\\
\hline
\end{tabular}
\end{table}

\section{Quality Control}\label{quality-control-1}

\subsection{Failed Samples}\label{failed-samples}

We use an aggregated \texttt{QCFlag} that considers a sample to have failed QC if any of the following conditions are true:

\begin{itemize}
\tightlist
\item
  \texttt{linFlag}: linearity of positive controls with positive control concentrations is less than 0.95, or linearity measures are unknown
\item
  \texttt{imagingFlag}: percent of field of view is less than 75\%
\item
  \texttt{spcFlag}: smallest positive control is less than the lower limit of detection (negative control average expression less two times the negative control standard deviation), or negative control average expression equals zero
\item
  \texttt{normFlag}: signal to noise ratio less than 100, or percent of genes detected is less than 50. Note: these thresholds were determined by examining the \hyperref[gd-vs.-snr]{\%GD vs.~SNR} relationship below.
\end{itemize}

\begin{table}

\caption{\label{tab:qc-failed}Quality Control Summary}
\centering
\begin{tabular}[t]{l|c|c|c}
\hline
**Quality Control Flag** & **CS1**, N = 294 & **CS2**, N = 903 & **CS3**, N = 2,477\\
\hline
\_\_Linearity\_\_ &  &  & \\
\hline
Failed & 0 (0\%) & 4 (0.4\%) & 0 (0\%)\\
\hline
Passed & 294 (100\%) & 899 (100\%) & 2,477 (100\%)\\
\hline
\_\_Imaging\_\_ &  &  & \\
\hline
Failed & 3 (1.0\%) & 0 (0\%) & 4 (0.2\%)\\
\hline
Passed & 291 (99\%) & 903 (100\%) & 2,473 (100\%)\\
\hline
\_\_Smallest Positive Control\_\_ &  &  & \\
\hline
Failed & 0 (0\%) & 2 (0.2\%) & 0 (0\%)\\
\hline
Passed & 294 (100\%) & 901 (100\%) & 2,477 (100\%)\\
\hline
\_\_Normality\_\_ &  &  & \\
\hline
Failed & 5 (1.7\%) & 19 (2.1\%) & 200 (8.1\%)\\
\hline
Passed & 289 (98\%) & 884 (98\%) & 2,277 (92\%)\\
\hline
\_\_Overall QC\_\_ &  &  & \\
\hline
Failed & 8 (2.7\%) & 21 (2.3\%) & 204 (8.2\%)\\
\hline
Passed & 286 (97\%) & 882 (98\%) & 2,273 (92\%)\\
\hline
\end{tabular}
\end{table}

\subsection{\%GD vs.~SNR}\label{gd-vs.-snr}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/qc-gd-snr-all-1} 

}

\caption{\% Genes Detected vs. Signal to Noise Ratio}\label{fig:qc-gd-snr-all}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/qc-gd-snr-zoomed-1} 

}

\caption{\% Genes Detected vs. Signal to Noise Ratio (Zoomed)}\label{fig:qc-gd-snr-zoomed}
\end{figure}

\section{Pairwise Gene Expression}\label{pairwise-gene-expression}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/genes-cs13-rand1-1} 

}

\caption{Random1-Normalized CS1 vs. CS3 Gene Expression}\label{fig:genes-cs13-rand1}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/genes-cs23-rand1-1} 

}

\caption{Random1-Normalized CS2 vs. CS3 Gene Expression}\label{fig:genes-cs23-rand1}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/genes-cs13-hkgenes-1} 

}

\caption{HKgenes-Normalized CS1 vs. CS3 Gene Expression}\label{fig:genes-cs13-hkgenes}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/genes-cs23-hkgenes-1} 

}

\caption{HKgenes-Normalized CS2 vs. CS3 Gene Expression}\label{fig:genes-cs23-hkgenes}
\end{figure}

\chapter{Results}\label{results}

We summarize cross-validated training performance of class metrics in the training set. The accuracy, F1-score, kappa, and G-mean are the metrics of interest. Workflows are ordered by their mean estimates across the outer folds of the nested CV for each metric.

\section{Training Set}\label{training-set}

\subsection{Accuracy}\label{accuracy-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-accuracy-1} 

}

\caption{Training Set Mean Accuracy}\label{fig:train-accuracy}
\end{figure}

\begin{table}

\caption{\label{tab:train-accuracy-table}Training Set Mean Accuracy}
\centering
\begin{tabular}[t]{l|l|l|l|l}
\hline
\multicolumn{1}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{2-5}
Subsampling & rf & svm & xgb & mr\\
\hline
none & 0.933 & \cellcolor[HTML]{90ee90}{0.938} & 0.818 & 0.936\\
\hline
down & 0.854 & 0.798 & 0.358 & 0.838\\
\hline
up & 0.93 & 0.929 & 0.934 & 0.911\\
\hline
smote & 0.937 & 0.934 & 0.936 & 0.914\\
\hline
hybrid & 0.931 & 0.932 & 0.926 & 0.901\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-accuracy-class-1} 

}

\caption{Training Set Class-Specific Mean Accuracy}\label{fig:train-accuracy-class}
\end{figure}

\begin{table}

\caption{\label{tab:train-accuracy-class-table}Training Set Class-Specific Mean Accuracy}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{3-6}
Subsampling & Histotype & rf & svm & xgb & mr\\
\hline
 & HGSC & 0.95 & 0.954 & 0.818 & 0.949\\
\cline{2-6}
 & CCOC & 0.982 & 0.98 & 0.939 & 0.981\\
\cline{2-6}
 & ENOC & 0.965 & 0.973 & 0.947 & 0.973\\
\cline{2-6}
 & LGSC & 0.984 & 0.983 & 0.982 & 0.982\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash none} & MUC & 0.986 & 0.986 & 0.951 & \cellcolor[HTML]{90ee90}{0.988}\\
\cline{1-6}
 & HGSC & 0.884 & 0.835 & 0.446 & 0.872\\
\cline{2-6}
 & CCOC & 0.975 & 0.973 & 0.935 & 0.972\\
\cline{2-6}
 & ENOC & 0.957 & 0.926 & 0.597 & 0.947\\
\cline{2-6}
 & LGSC & 0.921 & 0.892 & 0.788 & 0.914\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash down} & MUC & 0.97 & 0.97 & 0.951 & 0.972\\
\cline{1-6}
 & HGSC & 0.948 & 0.952 & 0.958 & 0.94\\
\cline{2-6}
 & CCOC & 0.98 & 0.973 & 0.981 & 0.977\\
\cline{2-6}
 & ENOC & 0.971 & 0.967 & 0.967 & 0.954\\
\cline{2-6}
 & LGSC & 0.977 & 0.982 & 0.982 & 0.973\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash up} & MUC & 0.984 & 0.984 & 0.981 & 0.977\\
\cline{1-6}
 & HGSC & 0.963 & 0.96 & 0.958 & 0.942\\
\cline{2-6}
 & CCOC & 0.977 & 0.973 & 0.979 & 0.975\\
\cline{2-6}
 & ENOC & 0.969 & 0.966 & 0.969 & 0.955\\
\cline{2-6}
 & LGSC & 0.98 & 0.983 & 0.982 & 0.975\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash smote} & MUC & 0.986 & 0.986 & 0.983 & 0.981\\
\cline{1-6}
 & HGSC & 0.955 & 0.955 & 0.951 & 0.928\\
\cline{2-6}
 & CCOC & 0.978 & 0.974 & 0.977 & 0.977\\
\cline{2-6}
 & ENOC & 0.969 & 0.966 & 0.965 & 0.955\\
\cline{2-6}
 & LGSC & 0.979 & 0.983 & 0.977 & 0.965\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash hybrid} & MUC & 0.98 & 0.985 & 0.982 & 0.977\\
\hline
\end{tabular}
\end{table}

\subsection{Sensitivity}\label{sensitivity-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-sens-1} 

}

\caption{Training Set Mean Sensitivity}\label{fig:train-sens}
\end{figure}

\begin{table}

\caption{\label{tab:train-sens-table}Training Set Mean Sensitivity}
\centering
\begin{tabular}[t]{l|l|l|l|l}
\hline
\multicolumn{1}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{2-5}
Subsampling & rf & svm & xgb & mr\\
\hline
none & 0.696 & 0.695 & 0.215 & 0.655\\
\hline
down & \cellcolor[HTML]{90ee90}{0.826} & 0.806 & 0.2 & 0.819\\
\hline
up & 0.676 & 0.745 & 0.771 & 0.82\\
\hline
smote & 0.714 & 0.746 & 0.788 & 0.818\\
\hline
hybrid & 0.727 & 0.773 & 0.764 & 0.816\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-sens-class-1} 

}

\caption{Training Set Class-Specific Mean Sensitivity}\label{fig:train-sens-class}
\end{figure}

\begin{table}

\caption{\label{tab:train-sens-class-table}Cross-Validated Training Set Class-Specific Mean Sensitivity}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{3-6}
Subsampling & Histotype & rf & svm & xgb & mr\\
\hline
 & HGSC & 0.991 & 0.992 & \cellcolor[HTML]{90ee90}{1} & 0.994\\
\cline{2-6}
 & CCOC & 0.827 & 0.779 & 0.074 & 0.782\\
\cline{2-6}
 & ENOC & 0.58 & 0.728 & 0 & 0.65\\
\cline{2-6}
 & LGSC & 0.25 & 0.19 & 0 & 0\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash none} & MUC & 0.833 & 0.786 & 0 & 0.849\\
\cline{1-6}
 & HGSC & 0.866 & 0.802 & 0.4 & 0.849\\
\cline{2-6}
 & CCOC & 0.863 & 0.789 & 0 & 0.829\\
\cline{2-6}
 & ENOC & 0.685 & 0.733 & 0.4 & 0.678\\
\cline{2-6}
 & LGSC & 0.87 & 0.95 & 0.2 & 0.91\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash down} & MUC & 0.845 & 0.754 & 0 & 0.831\\
\cline{1-6}
 & HGSC & 0.982 & 0.978 & 0.97 & 0.935\\
\cline{2-6}
 & CCOC & 0.827 & 0.747 & 0.84 & 0.839\\
\cline{2-6}
 & ENOC & 0.652 & 0.675 & 0.726 & 0.708\\
\cline{2-6}
 & LGSC & 0.107 & 0.552 & 0.44 & 0.76\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash up} & MUC & 0.814 & 0.771 & 0.877 & 0.858\\
\cline{1-6}
 & HGSC & 0.984 & 0.978 & 0.971 & 0.94\\
\cline{2-6}
 & CCOC & 0.827 & 0.799 & 0.874 & 0.826\\
\cline{2-6}
 & ENOC & 0.72 & 0.701 & 0.666 & 0.691\\
\cline{2-6}
 & LGSC & 0.207 & 0.452 & 0.552 & 0.76\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash smote} & MUC & 0.833 & 0.801 & 0.877 & 0.873\\
\cline{1-6}
 & HGSC & 0.971 & 0.97 & 0.959 & 0.923\\
\cline{2-6}
 & CCOC & 0.841 & 0.809 & 0.848 & 0.837\\
\cline{2-6}
 & ENOC & 0.732 & 0.705 & 0.709 & 0.717\\
\cline{2-6}
 & LGSC & 0.229 & 0.581 & 0.424 & 0.743\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash hybrid} & MUC & 0.862 & 0.801 & 0.877 & 0.858\\
\hline
\end{tabular}
\end{table}

\subsection{Specificity}\label{specificity-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-spec-1} 

}

\caption{Training Set Mean Specificity}\label{fig:train-spec}
\end{figure}

\begin{table}

\caption{\label{tab:train-spec-table}Training Set Mean Specificity}
\centering
\begin{tabular}[t]{l|l|l|l|l}
\hline
\multicolumn{1}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{2-5}
Subsampling & rf & svm & xgb & mr\\
\hline
none & 0.949 & 0.952 & 0.804 & 0.946\\
\hline
down & 0.964 & 0.954 & 0.8 & 0.961\\
\hline
up & 0.952 & 0.958 & 0.971 & \cellcolor[HTML]{90ee90}{0.974}\\
\hline
smote & 0.966 & 0.966 & 0.97 & 0.973\\
\hline
hybrid & 0.966 & 0.967 & 0.97 & 0.972\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-spec-class-1} 

}

\caption{Training Set Class-Specific Mean Specificity}\label{fig:train-spec-class}
\end{figure}

\begin{table}

\caption{\label{tab:train-spec-class-table}Cross-Validated Training Set Class-Specific Mean Specificity}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{3-6}
Subsampling & Histotype & rf & svm & xgb & mr\\
\hline
 & HGSC & 0.772 & 0.786 & 0.022 & 0.751\\
\cline{2-6}
 & CCOC & 0.994 & 0.995 & \cellcolor[HTML]{90ee90}{1} & 0.996\\
\cline{2-6}
 & ENOC & 0.987 & 0.987 & \cellcolor[HTML]{90ee90}{1} & 0.991\\
\cline{2-6}
 & LGSC & \cellcolor[HTML]{90ee90}{1} & 0.999 & \cellcolor[HTML]{90ee90}{1} & \cellcolor[HTML]{90ee90}{1}\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash none} & MUC & 0.993 & 0.995 & \cellcolor[HTML]{90ee90}{1} & 0.994\\
\cline{1-6}
 & HGSC & 0.965 & 0.978 & 0.6 & 0.97\\
\cline{2-6}
 & CCOC & 0.984 & 0.986 & \cellcolor[HTML]{90ee90}{1} & 0.983\\
\cline{2-6}
 & ENOC & 0.973 & 0.937 & 0.6 & 0.963\\
\cline{2-6}
 & LGSC & 0.922 & 0.891 & 0.8 & 0.914\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash down} & MUC & 0.976 & 0.981 & \cellcolor[HTML]{90ee90}{1} & 0.979\\
\cline{1-6}
 & HGSC & 0.795 & 0.832 & 0.904 & 0.956\\
\cline{2-6}
 & CCOC & 0.991 & 0.99 & 0.991 & 0.987\\
\cline{2-6}
 & ENOC & 0.99 & 0.984 & 0.98 & 0.969\\
\cline{2-6}
 & LGSC & 0.993 & 0.99 & 0.992 & 0.977\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash up} & MUC & 0.992 & 0.994 & 0.986 & 0.982\\
\cline{1-6}
 & HGSC & 0.87 & 0.877 & 0.899 & 0.948\\
\cline{2-6}
 & CCOC & 0.989 & 0.985 & 0.987 & 0.986\\
\cline{2-6}
 & ENOC & 0.983 & 0.982 & 0.986 & 0.97\\
\cline{2-6}
 & LGSC & 0.994 & 0.992 & 0.99 & 0.979\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash smote} & MUC & 0.993 & 0.995 & 0.988 & 0.985\\
\cline{1-6}
 & HGSC & 0.882 & 0.885 & 0.913 & 0.953\\
\cline{2-6}
 & CCOC & 0.989 & 0.986 & 0.987 & 0.987\\
\cline{2-6}
 & ENOC & 0.983 & 0.981 & 0.98 & 0.968\\
\cline{2-6}
 & LGSC & 0.993 & 0.989 & 0.986 & 0.968\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash hybrid} & MUC & 0.986 & 0.993 & 0.986 & 0.982\\
\hline
\end{tabular}
\end{table}

\subsection{F1-Score}\label{f1-score-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-f1-1} 

}

\caption{Training Set Mean F1-Score}\label{fig:train-f1}
\end{figure}

\begin{table}

\caption{\label{tab:train-f1-table}Training Set Mean F1-Score}
\centering
\begin{tabular}[t]{l|l|l|l|l}
\hline
\multicolumn{1}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{2-5}
Subsampling & rf & svm & xgb & mr\\
\hline
none & \cellcolor[HTML]{FFD700}{0.819} & \cellcolor[HTML]{FFD700}{0.799} & \cellcolor[HTML]{FF0000}{0.78} & \cellcolor[HTML]{FF0000}{0.851}\\
\hline
down & 0.674 & 0.622 & \cellcolor[HTML]{FF0000}{0.415} & 0.659\\
\hline
up & 0.7 & 0.749 & 0.752 & 0.723\\
\hline
smote & 0.724 & 0.747 & \cellcolor[HTML]{90ee90}{0.771} & 0.731\\
\hline
hybrid & 0.717 & 0.76 & 0.731 & 0.716\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-f1-class-1} 

}

\caption{Training Set Class-Specific Mean F1-Score}\label{fig:train-f1-class}
\end{figure}

\begin{table}

\caption{\label{tab:train-f1-class-table}Cross-Validated Training Set Class-Specific Mean F1-Score}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{3-6}
Subsampling & Histotype & rf & svm & xgb & mr\\
\hline
 & HGSC & \cellcolor[HTML]{FFD700}{0.97} & \cellcolor[HTML]{FFD700}{0.972} & \cellcolor[HTML]{FF0000}{0.9} & \cellcolor[HTML]{FF0000}{0.969}\\
\cline{2-6}
 & CCOC & \cellcolor[HTML]{FFD700}{0.861} & \cellcolor[HTML]{FFD700}{0.836} & \cellcolor[HTML]{FF0000}{0.31} & \cellcolor[HTML]{FF0000}{0.846}\\
\cline{2-6}
 & ENOC & \cellcolor[HTML]{FFD700}{0.641} & \cellcolor[HTML]{FFD700}{0.741} & \cellcolor[HTML]{FF0000}{NaN} & \cellcolor[HTML]{FF0000}{0.72}\\
\cline{2-6}
 & LGSC & \cellcolor[HTML]{FFD700}{0.7} & \cellcolor[HTML]{FFD700}{0.444} & \cellcolor[HTML]{FF0000}{NaN} & \cellcolor[HTML]{FF0000}{NaN}\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash none} & MUC & \cellcolor[HTML]{FFD700}{0.848} & \cellcolor[HTML]{FFD700}{0.834} & \cellcolor[HTML]{FF0000}{NaN} & \cellcolor[HTML]{FF0000}{0.869}\\
\cline{1-6}
 & HGSC & 0.924 & 0.885 & \cellcolor[HTML]{FF0000}{0.904} & 0.914\\
\cline{2-6}
 & CCOC & 0.827 & 0.788 & \cellcolor[HTML]{FF0000}{NaN} & 0.8\\
\cline{2-6}
 & ENOC & 0.616 & 0.505 & \cellcolor[HTML]{FF0000}{0.117} & 0.579\\
\cline{2-6}
 & LGSC & 0.279 & 0.24 & \cellcolor[HTML]{FF0000}{0.032} & 0.275\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash down} & MUC & 0.722 & 0.692 & \cellcolor[HTML]{FF0000}{NaN} & 0.725\\
\cline{1-6}
 & HGSC & 0.968 & 0.971 & 0.974 & 0.962\\
\cline{2-6}
 & CCOC & 0.847 & 0.786 & 0.856 & 0.829\\
\cline{2-6}
 & ENOC & 0.71 & 0.669 & 0.699 & 0.608\\
\cline{2-6}
 & LGSC & 0.147 & 0.498 & 0.425 & 0.448\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash up} & MUC & 0.828 & 0.82 & 0.807 & 0.766\\
\cline{1-6}
 & HGSC & \cellcolor[HTML]{90ee90}{0.977} & 0.975 & 0.974 & 0.963\\
\cline{2-6}
 & CCOC & 0.831 & 0.788 & 0.852 & 0.816\\
\cline{2-6}
 & ENOC & 0.703 & 0.673 & 0.686 & 0.612\\
\cline{2-6}
 & LGSC & 0.259 & 0.457 & 0.516 & 0.469\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash smote} & MUC & 0.847 & 0.844 & 0.829 & 0.794\\
\cline{1-6}
 & HGSC & 0.972 & 0.972 & 0.969 & 0.954\\
\cline{2-6}
 & CCOC & 0.838 & 0.803 & 0.835 & 0.829\\
\cline{2-6}
 & ENOC & 0.708 & 0.676 & 0.669 & 0.623\\
\cline{2-6}
 & LGSC & 0.273 & 0.516 & 0.362 & 0.406\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash hybrid} & MUC & 0.795 & 0.833 & 0.819 & 0.767\\
\hline
\end{tabular}
\end{table}

\subsection{Balanced Accuracy}\label{balanced-accuracy-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-bal-accuracy-1} 

}

\caption{Training Set Mean Balanced Accuracy}\label{fig:train-bal-accuracy}
\end{figure}

\begin{table}

\caption{\label{tab:train-bal-accuracy-table}Training Set Mean Balanced Accuracy}
\centering
\begin{tabular}[t]{l|l|l|l|l}
\hline
\multicolumn{1}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{2-5}
Subsampling & rf & svm & xgb & mr\\
\hline
none & 0.823 & 0.824 & 0.51 & 0.801\\
\hline
down & 0.895 & 0.88 & 0.5 & 0.89\\
\hline
up & 0.814 & 0.851 & 0.871 & \cellcolor[HTML]{90ee90}{0.897}\\
\hline
smote & 0.84 & 0.856 & 0.879 & 0.896\\
\hline
hybrid & 0.847 & 0.87 & 0.867 & 0.894\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-bal-accuracy-class-1} 

}

\caption{Training Set Class-Specific Mean Balanced Accuracy}\label{fig:train-bal-accuracy-class}
\end{figure}

\begin{table}

\caption{\label{tab:train-bal-accuracy-class-table}Training Set Class-Specific Mean Balanced Accuracy}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{3-6}
Subsampling & Histotype & rf & svm & xgb & mr\\
\hline
 & HGSC & 0.881 & 0.889 & 0.511 & 0.873\\
\cline{2-6}
 & CCOC & 0.91 & 0.887 & 0.537 & 0.889\\
\cline{2-6}
 & ENOC & 0.783 & 0.858 & 0.5 & 0.821\\
\cline{2-6}
 & LGSC & 0.625 & 0.595 & 0.5 & 0.5\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash none} & MUC & 0.913 & 0.891 & 0.5 & 0.921\\
\cline{1-6}
 & HGSC & 0.915 & 0.89 & 0.5 & 0.909\\
\cline{2-6}
 & CCOC & 0.923 & 0.888 & 0.5 & 0.906\\
\cline{2-6}
 & ENOC & 0.829 & 0.835 & 0.5 & 0.82\\
\cline{2-6}
 & LGSC & 0.896 & 0.92 & 0.5 & 0.912\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash down} & MUC & 0.911 & 0.867 & 0.5 & 0.905\\
\cline{1-6}
 & HGSC & 0.889 & 0.905 & 0.937 & \cellcolor[HTML]{90ee90}{0.946}\\
\cline{2-6}
 & CCOC & 0.909 & 0.869 & 0.916 & 0.913\\
\cline{2-6}
 & ENOC & 0.821 & 0.83 & 0.853 & 0.838\\
\cline{2-6}
 & LGSC & 0.55 & 0.771 & 0.716 & 0.869\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash up} & MUC & 0.903 & 0.883 & 0.932 & 0.92\\
\cline{1-6}
 & HGSC & 0.927 & 0.928 & 0.935 & 0.944\\
\cline{2-6}
 & CCOC & 0.908 & 0.892 & 0.931 & 0.906\\
\cline{2-6}
 & ENOC & 0.851 & 0.841 & 0.826 & 0.831\\
\cline{2-6}
 & LGSC & 0.6 & 0.722 & 0.771 & 0.869\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash smote} & MUC & 0.913 & 0.898 & 0.933 & 0.929\\
\cline{1-6}
 & HGSC & 0.927 & 0.928 & 0.936 & 0.938\\
\cline{2-6}
 & CCOC & 0.915 & 0.898 & 0.918 & 0.912\\
\cline{2-6}
 & ENOC & 0.858 & 0.843 & 0.844 & 0.843\\
\cline{2-6}
 & LGSC & 0.611 & 0.785 & 0.705 & 0.856\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash hybrid} & MUC & 0.924 & 0.897 & 0.932 & 0.92\\
\hline
\end{tabular}
\end{table}

\subsection{Kappa}\label{kappa-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-kappa-1} 

}

\caption{Training Set Mean Kappa}\label{fig:train-kappa}
\end{figure}

\begin{table}

\caption{\label{tab:train-kappa-table}Training Set Mean Kappa}
\centering
\begin{tabular}[t]{l|l|l|l|l}
\hline
\multicolumn{1}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{2-5}
Subsampling & rf & svm & xgb & mr\\
\hline
none & 0.776 & 0.793 & 0.038 & 0.782\\
\hline
down & 0.642 & 0.558 & 0 & 0.618\\
\hline
up & 0.772 & 0.775 & 0.802 & 0.754\\
\hline
smote & 0.803 & 0.794 & \cellcolor[HTML]{90ee90}{0.808} & 0.761\\
\hline
hybrid & 0.788 & 0.792 & 0.784 & 0.734\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-kappa-class-1} 

}

\caption{Training Set Class-Specific Mean Kappa}\label{fig:train-kappa-class}
\end{figure}

\begin{table}

\caption{\label{tab:train-kappa-class-table}Training Set Class-Specific Mean Kappa}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{3-6}
Subsampling & Histotype & rf & svm & xgb & mr\\
\hline
 & HGSC & 0.821 & 0.835 & 0.035 & 0.814\\
\cline{2-6}
 & CCOC & 0.852 & 0.826 & 0.119 & 0.836\\
\cline{2-6}
 & ENOC & 0.622 & 0.727 & 0 & 0.706\\
\cline{2-6}
 & LGSC & 0.279 & 0.264 & 0 & 0\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash none} & MUC & 0.84 & 0.827 & 0 & 0.863\\
\cline{1-6}
 & HGSC & 0.686 & 0.598 & 0 & 0.665\\
\cline{2-6}
 & CCOC & 0.813 & 0.773 & 0 & 0.785\\
\cline{2-6}
 & ENOC & 0.594 & 0.47 & 0 & 0.552\\
\cline{2-6}
 & LGSC & 0.258 & 0.216 & 0 & 0.252\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash down} & MUC & 0.707 & 0.676 & 0 & 0.711\\
\cline{1-6}
 & HGSC & 0.818 & 0.834 & 0.864 & 0.819\\
\cline{2-6}
 & CCOC & 0.836 & 0.773 & 0.846 & 0.817\\
\cline{2-6}
 & ENOC & 0.696 & 0.652 & 0.681 & 0.584\\
\cline{2-6}
 & LGSC & 0.137 & 0.489 & 0.416 & 0.437\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash up} & MUC & 0.82 & 0.812 & 0.797 & 0.755\\
\cline{1-6}
 & HGSC & \cellcolor[HTML]{90ee90}{0.875} & 0.865 & 0.865 & 0.824\\
\cline{2-6}
 & CCOC & 0.819 & 0.773 & 0.84 & 0.803\\
\cline{2-6}
 & ENOC & 0.687 & 0.655 & 0.67 & 0.588\\
\cline{2-6}
 & LGSC & 0.25 & 0.449 & 0.508 & 0.458\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash smote} & MUC & 0.839 & 0.836 & 0.82 & 0.784\\
\cline{1-6}
 & HGSC & 0.852 & 0.851 & 0.845 & 0.789\\
\cline{2-6}
 & CCOC & 0.826 & 0.79 & 0.823 & 0.816\\
\cline{2-6}
 & ENOC & 0.692 & 0.659 & 0.651 & 0.599\\
\cline{2-6}
 & LGSC & 0.264 & 0.507 & 0.351 & 0.391\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash hybrid} & MUC & 0.784 & 0.825 & 0.809 & 0.756\\
\hline
\end{tabular}
\end{table}

\subsection{G-mean}\label{g-mean}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-gmean-1} 

}

\caption{Training Set Mean G-mean}\label{fig:train-gmean}
\end{figure}

\begin{table}

\caption{\label{tab:train-gmean-table}Training Set Mean G-mean}
\centering
\begin{tabular}[t]{l|l|l|l|l}
\hline
\multicolumn{1}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{2-5}
Subsampling & rf & svm & xgb & mr\\
\hline
none & 0.293 & 0.389 & 0 & 0\\
\hline
down & \cellcolor[HTML]{90ee90}{0.815} & 0.795 & 0 & 0.81\\
\hline
up & 0.246 & 0.71 & 0.602 & 0.666\\
\hline
smote & 0.403 & 0.709 & 0.749 & 0.665\\
\hline
hybrid & 0.415 & 0.747 & 0.598 & 0.794\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/train-gmean-class-1} 

}

\caption{Training Set Class-Specific Mean G-mean}\label{fig:train-gmean-class}
\end{figure}

\begin{table}

\caption{\label{tab:train-gmean-class-table}Training Set Class-Specific Mean G-mean}
\centering
\begin{tabular}[t]{l|l|l|l|l|l}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{4}{c}{Algorithms} \\
\cline{3-6}
Subsampling & Histotype & rf & svm & xgb & mr\\
\hline
 & HGSC & 0.873 & 0.882 & 0.094 & 0.863\\
\cline{2-6}
 & CCOC & 0.905 & 0.878 & 0.171 & 0.88\\
\cline{2-6}
 & ENOC & 0.753 & 0.844 & 0 & 0.797\\
\cline{2-6}
 & LGSC & 0.3 & 0.331 & 0 & 0\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash none} & MUC & 0.909 & 0.88 & 0 & 0.916\\
\cline{1-6}
 & HGSC & 0.914 & 0.884 & 0 & 0.907\\
\cline{2-6}
 & CCOC & 0.92 & 0.88 & 0 & 0.9\\
\cline{2-6}
 & ENOC & 0.811 & 0.824 & 0 & 0.803\\
\cline{2-6}
 & LGSC & 0.891 & 0.918 & 0 & 0.909\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash down} & MUC & 0.907 & 0.857 & 0 & 0.901\\
\cline{1-6}
 & HGSC & 0.883 & 0.901 & 0.936 & \cellcolor[HTML]{90ee90}{0.946}\\
\cline{2-6}
 & CCOC & 0.904 & 0.858 & 0.911 & 0.908\\
\cline{2-6}
 & ENOC & 0.798 & 0.81 & 0.842 & 0.823\\
\cline{2-6}
 & LGSC & 0.204 & 0.715 & 0.566 & 0.769\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash up} & MUC & 0.897 & 0.872 & 0.928 & 0.916\\
\cline{1-6}
 & HGSC & 0.925 & 0.926 & 0.934 & 0.944\\
\cline{2-6}
 & CCOC & 0.903 & 0.885 & 0.928 & 0.902\\
\cline{2-6}
 & ENOC & 0.836 & 0.824 & 0.805 & 0.816\\
\cline{2-6}
 & LGSC & 0.346 & 0.657 & 0.715 & 0.77\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash smote} & MUC & 0.909 & 0.89 & 0.93 & 0.925\\
\cline{1-6}
 & HGSC & 0.926 & 0.926 & 0.936 & 0.937\\
\cline{2-6}
 & CCOC & 0.91 & 0.892 & 0.914 & 0.908\\
\cline{2-6}
 & ENOC & 0.845 & 0.826 & 0.826 & 0.831\\
\cline{2-6}
 & LGSC & 0.363 & 0.744 & 0.568 & 0.828\\
\cline{2-6}
\multirow{-5}{*}{\raggedright\arraybackslash hybrid} & MUC & 0.92 & 0.889 & 0.929 & 0.916\\
\hline
\end{tabular}
\end{table}

\section{Rank Aggregation}\label{rank-aggregation}

\begin{center}\includegraphics{OV_Histotypes_RSF_files/figure-latex/per-class-metrics-1} \end{center}

The 19 workflows are ordered in the table by their aggregated ranks using the Genetic Algorithm. We see that the best performing methods involve the sequential and two-step algorithms.

\subsection{Top Workflows}\label{top-workflows}

We look at the per-class evaluation metrics of the top 4 workflows.

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/metrics-top4-1} 

}

\caption{Top 4 Workflow Per-Class Evaluation Metrics}\label{fig:metrics-top4}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/metrics-f1-top4-1} 

}

\caption{Top 4 Workflow Per-Class F1-Scores}\label{fig:metrics-f1-top4}
\end{figure}

Misclassified cases from a previous step of the sequence of classifiers are not included in subsequent steps of the training set CV folds. Thus, we cannot piece together the test set predictions from the sequential and two-step algorithms to obtain overall metrics.

\section{Optimal Gene Sets}\label{optimal-gene-sets}

\subsection{Sequential Algorithm}\label{sequential-algorithm-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/gene-seq-1} 

}

\caption{Gene Optimization for Sequential Classifier}\label{fig:gene-seq}
\end{figure}

In the sequential algorithm, sequences 1, 2, and 4 have relatively flat average F1-scores across the number of genes added. However, we can observe in sequence 3, the F1-score stabilizes at around 0.91 when we reach 12 genes added, hence the optimal number of genes used will be n=28+12=40 The added genes are: CYP2C18, TFF3, HNF1B, SLC3A1, MAP1LC3A, IL6, BRCA1, EGFL6, IGFBP1, MUC5B, BCL2 and WT1.

\subsection{Two-Step Algorithm}\label{two-step-algorithm-1}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/gene-2s-1} 

}

\caption{Gene Optimization for Two-Step Classifier}\label{fig:gene-2s}
\end{figure}

Since the second step of the classifier fits a multinomial model, we use the macro F1-score as the measure to analyze gene entry. In the two-step classifier, we see that in Step 2, the F1-score stabilizes at around 0.84 when we reach 11 added. The optimal number of genes used will be n=28+11=39. The added genes are: CYP2C18, MUC5B, HNF1B, SLC3A1, EGFL6, WT1, TFF3, MET, IGFBP1, TP53 and KLK7.

\section{Test Set Performance}\label{test-set-performance}

Now we'd like to see how our best methods perform in the confirmation and validation sets. The class-specific F1-scores will be used.

The top 2 methods are:

\begin{itemize}
\tightlist
\item
  \textbf{sequential}: sequential algorithm with hybrid subsampling at every step. The sequence of algorithms used are:

  \begin{itemize}
  \tightlist
  \item
    HGSC vs.~non-HGSC using random forest
  \item
    CCOC vs.~non-CCOC using XGBoost
  \item
    ENOC vs.~non-ENOC using support vector machine
  \item
    LGSC vs.~MUC using support vector machine
  \end{itemize}
\item
  \textbf{two\_step}: two-step algorithm with hybrid subsampling at both steps. The sequence of algorithms used are:

  \begin{itemize}
  \tightlist
  \item
    HGSC vs.~non-HGSC using random forest
  \item
    CCOC vs.~ENOC vs.~MUC vs.~LGSC support vector machine
  \end{itemize}
\end{itemize}

We can test 2 additional methods by using either the full set of genes or the optimal set of genes for both of these methods.

\subsection{Confirmation Set}\label{confirmation-set}

\begin{table}

\caption{\label{tab:conf-eval-overall}Overall Evaluation Metrics on Confirmation Set Models}
\centering
\begin{tabular}[t]{l|r|r|r|r|r|r|r}
\hline
Method & accuracy & sensitivity & specificity & f1 & bal\_accuracy & kappa & gmean\\
\hline
sequential\_full & 0.832 & 0.646 & 0.927 & 0.664 & 0.786 & 0.660 & 0.599\\
\hline
sequential\_optimal & 0.829 & 0.609 & 0.924 & 0.628 & 0.767 & 0.651 & 0.516\\
\hline
two\_step\_full & 0.838 & 0.674 & 0.928 & 0.694 & 0.801 & 0.673 & 0.638\\
\hline
two\_step\_optimal & 0.832 & 0.672 & 0.923 & 0.694 & 0.798 & 0.655 & 0.639\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/conf-conf-mat-1} 

}

\caption{Confusion Matrices for Confirmation Set Models}\label{fig:conf-conf-mat}
\end{figure}

\begin{table}

\caption{\label{tab:conf-eval-per-class}Per-Class Evaluation Metrics on Confirmation Set Model}
\centering
\begin{tabular}[t]{l|l|r|r|r|r|r}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{5}{c}{Histotypes} \\
\cline{3-7}
Method & Metric & HGSC & CCOC & ENOC & LGSC & MUC\\
\hline
 & accuracy & 0.866 & 0.961 & 0.894 & 0.975 & 0.980\\
\cline{2-7}
 & sensitivity & 0.948 & 0.853 & 0.481 & 0.385 & 0.704\\
\cline{2-7}
 & specificity & 0.710 & 0.975 & 0.976 & 0.987 & 0.992\\
\cline{2-7}
 & f1 & 0.903 & 0.837 & 0.600 & 0.385 & 0.745\\
\cline{2-7}
 & bal\_accuracy & 0.829 & 0.914 & 0.728 & 0.686 & 0.848\\
\cline{2-7}
 & kappa & 0.689 & 0.815 & 0.543 & 0.372 & 0.735\\
\cline{2-7}
\multirow{-7}{*}{\raggedright\arraybackslash two\_step\_full} & gmean & 0.821 & 0.912 & 0.685 & 0.616 & 0.835\\
\cline{1-7}
 & accuracy & 0.860 & 0.963 & 0.885 & 0.977 & 0.980\\
\cline{2-7}
 & sensitivity & 0.953 & 0.853 & 0.425 & 0.462 & 0.667\\
\cline{2-7}
 & specificity & 0.683 & 0.977 & 0.976 & 0.987 & 0.994\\
\cline{2-7}
 & f1 & 0.899 & 0.842 & 0.549 & 0.444 & 0.735\\
\cline{2-7}
 & bal\_accuracy & 0.818 & 0.915 & 0.700 & 0.724 & 0.830\\
\cline{2-7}
 & kappa & 0.672 & 0.821 & 0.489 & 0.433 & 0.724\\
\cline{2-7}
\multirow{-7}{*}{\raggedright\arraybackslash two\_step\_optimal} & gmean & 0.807 & 0.913 & 0.644 & 0.675 & 0.814\\
\cline{1-7}
 & accuracy & 0.866 & 0.963 & 0.890 & 0.975 & 0.970\\
\cline{2-7}
 & sensitivity & 0.948 & 0.827 & 0.481 & 0.308 & 0.667\\
\cline{2-7}
 & specificity & 0.710 & 0.981 & 0.970 & 0.989 & 0.984\\
\cline{2-7}
 & f1 & 0.903 & 0.838 & 0.590 & 0.333 & 0.655\\
\cline{2-7}
 & bal\_accuracy & 0.829 & 0.904 & 0.726 & 0.648 & 0.825\\
\cline{2-7}
 & kappa & 0.689 & 0.817 & 0.530 & 0.321 & 0.639\\
\cline{2-7}
\multirow{-7}{*}{\raggedright\arraybackslash sequential\_full} & gmean & 0.821 & 0.900 & 0.683 & 0.552 & 0.810\\
\cline{1-7}
 & accuracy & 0.863 & 0.964 & 0.890 & 0.974 & 0.967\\
\cline{2-7}
 & sensitivity & 0.950 & 0.840 & 0.472 & 0.154 & 0.630\\
\cline{2-7}
 & specificity & 0.697 & 0.981 & 0.972 & 0.990 & 0.982\\
\cline{2-7}
 & f1 & 0.901 & 0.846 & 0.585 & 0.190 & 0.618\\
\cline{2-7}
 & bal\_accuracy & 0.824 & 0.910 & 0.722 & 0.572 & 0.806\\
\cline{2-7}
 & kappa & 0.681 & 0.825 & 0.525 & 0.178 & 0.601\\
\cline{2-7}
\multirow{-7}{*}{\raggedright\arraybackslash sequential\_optimal} & gmean & 0.814 & 0.908 & 0.677 & 0.390 & 0.786\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/roc-curve-seq-full-1} 

}

\caption{ROC Curves for Sequential Full Model in Confirmation Set}\label{fig:roc-curve-seq-full}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/roc-curve-seq-opt-1} 

}

\caption{ROC Curves for Sequential Optimal Model in Confirmation Set}\label{fig:roc-curve-seq-opt}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/roc-curve-two-step-full-1} 

}

\caption{ROC Curves for Two-Step Full Model in Confirmation Set}\label{fig:roc-curve-two-step-full}
\end{figure}

\begin{figure}[H]

{\centering \includegraphics[width=1\linewidth]{OV_Histotypes_RSF_files/figure-latex/roc-curve-two-step-opt-1} 

}

\caption{ROC Curves for Two-Step Optimal Model in Confirmation Set}\label{fig:roc-curve-two-step-opt}
\end{figure}

\subsection{Validation Set}\label{validation-set}

\begin{table}

\caption{\label{tab:val-eval-overall}Overall Evaluation Metrics on Validation Set Model}
\centering
\begin{tabular}[t]{l|r|r|r|r|r|r|r}
\hline
Method & accuracy & sensitivity & specificity & f1 & bal\_accuracy & kappa & gmean\\
\hline
two\_step\_optimal & 0.87 & 0.713 & 0.94 & 0.679 & 0.826 & 0.691 & 0.662\\
\hline
\end{tabular}
\end{table}

\begin{figure}[H]

{\centering \includegraphics{OV_Histotypes_RSF_files/figure-latex/val-conf-mat-1} 

}

\caption{Confusion Matrix for Validation Set Model}\label{fig:val-conf-mat}
\end{figure}

\begin{table}

\caption{\label{tab:val-eval-per-class}Per-Class Eevaluation Metrics on Validation Set Model}
\centering
\begin{tabular}[t]{l|l|r|r|r|r|r}
\hline
\multicolumn{2}{c|}{ } & \multicolumn{5}{c}{Histotypes} \\
\cline{3-7}
Method & Metric & HGSC & CCOC & ENOC & LGSC & MUC\\
\hline
 & accuracy & 0.894 & 0.972 & 0.928 & 0.974 & 0.972\\
\cline{2-7}
 & sensitivity & 0.934 & 0.937 & 0.514 & 0.333 & 0.846\\
\cline{2-7}
 & specificity & 0.776 & 0.975 & 0.984 & 0.987 & 0.976\\
\cline{2-7}
 & f1 & 0.929 & 0.855 & 0.628 & 0.343 & 0.638\\
\cline{2-7}
 & bal\_accuracy & 0.855 & 0.956 & 0.749 & 0.660 & 0.911\\
\cline{2-7}
 & kappa & 0.718 & 0.840 & 0.590 & 0.330 & 0.624\\
\cline{2-7}
\multirow{-7}{*}{\raggedright\arraybackslash two\_step\_optimal} & gmean & 0.852 & 0.956 & 0.711 & 0.574 & 0.909\\
\hline
\end{tabular}
\end{table}

  \bibliography{book.bib,packages.bib}
\addcontentsline{toc}{chapter}{\bibname}

\end{document}
